interface/views.py

```diff
@@ -296,3 +296,16 @@ def subs_for_user(request, course_pk, assignment_pk, username):
         'assignment': assignment,
         'submissions': submissions,
     })
+
+
+@login_required
+def user_page(request, username):
+    user = get_object_or_404(User, username=username)
+    submissions_aux = Submission.objects.all().order_by('-timestamp')
+    submissions = submissions_aux.filter(user=user)
+    if request.user.username != username:
+        log.warning(f'User attempted to access {username}')
+        raise Http404("You are not allowed to access this page.")
```

Comment:
I think we should have this `if` after line 303. There is no need for a submission querry if the user is not valid